src/middlewared/middlewared/plugins/account_/privilege.py

```diff
@@ -0,0 +1,161 @@
+import enum
+import errno
+
+from middlewared.schema import accepts, Bool, Dict, Int, List, Ref, Str, Patch
+from middlewared.service import CallError, CRUDService, private, ValidationErrors
+import middlewared.sqlalchemy as sa
+
+
+class BuiltinPrivileges(enum.Enum):
+    LOCAL_ADMINISTRATOR = "LOCAL_ADMINISTRATOR"
+
+
+class PrivilegeModel(sa.Model):
+    __tablename__ = "account_privilege"
+
+    id = sa.Column(sa.Integer(), primary_key=True)
+    builtin_name = sa.Column(sa.String(200), nullable=True)
+    name = sa.Column(sa.String(200))
+    local_groups = sa.Column(sa.JSON(type=list))
+    ds_groups = sa.Column(sa.JSON(type=list))
+    allowlist = sa.Column(sa.JSON(type=list))
+    web_shell = sa.Column(sa.Boolean())
+
+
+class PrivilegeService(CRUDService):
+
+    keys = {}
+
+    class Config:
+        namespace = "privilege"
+        datastore = "account.privilege"
+        datastore_extend = "privilege.item_extend"
+        datastore_extend_context = "privilege.item_extend_context"
+        cli_namespace = "auth.privilege"
+
+    ENTRY = Dict(
+        "privilege_entry",
+        Int("id"),
+        Str("builtin_name", null=True),
+        Str("name", required=True, empty=False),
+        List("local_groups", items=[Int("local_group")]),
+        List("ds_groups", items=[Int("ds_group")]),
+        List("allowlist", items=[Ref("allowlist_item")]),
+        Bool("web_shell", required=True),
+    )
+
+    @private
+    async def item_extend_context(self, rows, extra):
+        return {
+            "local_groups": {
```

Comment:
@anodos325 we probably want the same but for DS groups since the UI might want to display DS group names instead of DS group IDs. Will that have a good performance?

Ui might also probably want DS group names listing/autocompletion, will we be able to provide that?